cmd/firmware-action/filesystem/filesystem.go

```diff
@@ -265,3 +270,31 @@ func AnyFileNewerThan(path string, givenTime time.Time) (bool, error) {
 	// If path is neither file nor directory
 	return false, err
 }
+
+// Filenamify converts a string to a valid and safe filename
+func Filenamify(name string) string {
```

Comment:
Might be worthwhile to take the name and extension as separate arguments to make sure the extension isn't being cut off